fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/identifiers/FedoraID.java

```diff
@@ -0,0 +1,197 @@
+     */
+    public boolean isDescription() {
+        return isNonRdfSourceDescription;
+    }
+
+    /**
+     * Is the identifier for a hash uri?
+     * @return true if full id referenced a hash uri.
+     */
+    public boolean isHashUri() {
+        return hashUri != null;
+    }
+
+    /**
+     * Get the hash uri.
+     * @return the hash uri from the id or null if none.
+     */
+    public String getHashUri() {
+        return hashUri;
+    }
+
+    /**
+     * Return the ID of the base resource for this request.
+     * @return the shorten id.
+     */
+    public String getResourceId() {
+        return id;
+    }
+
+    /**
+     * Return the original full ID.
+     * @return the id.
+     */
+    public String getFullId() {
+        return fullId;
+    }
+
+    /**
+     * Return the Memento datetime as Instant.
+     * @return The datetime or null if not a memento.
+     */
+    public Instant getMementoInstant() {
+        return mementoDatetime;
+    }
+
+    /**
+     * Return the Memento datetime string.
+     * @return The yyyymmddhhiiss memento datetime or null if not a Memento.
+     */
+    public String getMementoString() {
+        return mementoDatetime_str;
+    }
+
+    /**
+     * Process the original ID into its parts without using a regular expression.
+     */
+    private void processIdentifier() {
+        // Regex pattern which decomposes a http resource uri into components
+        // The first group determines if it is an fcr:metadata non-rdf source.
+        // The second group determines if the path is for a memento or timemap.
+        // The third group allows for a memento identifier.
+        // The fourth group for allows ACL.
+        // The fifth group allows for any hashed suffixes.
+        // ".*?(/" + FCR_METADATA + ")?(/" + FCR_VERSIONS + "(/\\d{14})?)?(/" + FCR_ACL + ")?(\\#\\S+)?$");
+        String processID = this.fullId;
+        if (processID.equals(FEDORA_ID_PREFIX)) {
+            this.isRepositoryRoot = true;
+            this.id = this.fullId;
+            // Root has no other possible endpoints, so short circuit out.
+            return;
+        }
+        if (processID.contains("#")) {
+            this.hashUri = processID.split("#")[1];
+            processID = processID.split("#")[0];
+        }
+        if (processID.contains(FCR_ACL)) {
+            this.isAcl = true;
+            processID = processID.split(FCR_ACL)[0];
+        }
+        if (processID.contains(FCR_VERSIONS)) {
+            if (processID.split(FCR_VERSIONS).length == 2) {
+                final String afterVersion = processID.split(FCR_VERSIONS)[1];
+                if (afterVersion.matches("/\\d{14}")) {
+                    this.isMemento = true;
+                    this.mementoDatetime_str = afterVersion.substring(1);
+                    this.mementoDatetime = Instant.from(MEMENTO_LABEL_FORMATTER.parse(this.mementoDatetime_str));
+                } else {
+                    this.isTimemap = true;
```

Comment:
What is this scenario? ..when there is something that follows `fcr:versions`... but not a memento identifier??

Reply:
I can remove it and we can try an eventual integration test to see what happens.